llvm/include/llvm/Analysis/VecFuncs.def

```diff
@@ -943,6 +953,16 @@ TLI_DEFINE_VECFUNC("hypotf", "armpl_vhypotq_f32", FIXED(4), NOMASK, "_ZGV_LLVM_N
 TLI_DEFINE_VECFUNC("hypot", "armpl_svhypot_f64_x",  SCALABLE(2), MASKED, "_ZGVsMxvv")
 TLI_DEFINE_VECFUNC("hypotf", "armpl_svhypot_f32_x", SCALABLE(4), MASKED, "_ZGVsMxvv")
 
+TLI_DEFINE_VECFUNC("ilogb", "armpl_vilogbq_f64", FIXED(2), NOMASK, "_ZGV_LLVM_N2v")
+TLI_DEFINE_VECFUNC("ilogbf", "armpl_vilogbq_f32", FIXED(4), NOMASK, "_ZGV_LLVM_N4v")
+TLI_DEFINE_VECFUNC("ilogb", "armpl_svilogb_f64_x",  SCALABLE(2), MASKED, "_ZGVsMxv")
+TLI_DEFINE_VECFUNC("ilogbf", "armpl_svilogb_f32_x", SCALABLE(4), MASKED, "_ZGVsMxv")
+
+TLI_DEFINE_VECFUNC("ldexp", "armpl_vldexpq_f64", FIXED(2), NOMASK, "_ZGV_LLVM_N2vv")
```

Comment:
Shouldn't the 2nd argument of ldexp(f) be a linear argument and not a vector?

Reply:
why do you think so?
the function is defined as: 
``
double  ldexp( double arg, int exp)
``
both arguments are passed by value, so from the vector function I am expecting to return a vector of results  where each element is multiplied by a different power of 2 (i.e different element of the input exp vector) .

Reply:
No, the amath header file suggests it is an integer vector.

Reply:
Oh, right. For some reason I thought exp would be the same value for each element of arg.

@mgabka this explains your concern then, that we have 32bits float/int vectors and is OK, but what would actually happen with double/int vectors.